pkg/operator/statefulset.go

```diff
@@ -0,0 +1,916 @@
+				Host:   "localhost:9090",
+				Path:   probePath,
+			}
+			handler.Exec = &v1.ExecAction{
+				Command: []string{
+					"sh",
+					"-c",
+					fmt.Sprintf(
+						`if [ -x "$(command -v curl)" ]; then exec %s; elif [ -x "$(command -v wget)" ]; then exec %s; else exit 1; fi`,
+						CurlProber(probeURL.String()),
+						WgetProber(probeURL.String()),
+					),
+				},
+			}
+			return handler
+		}
+
+		handler.HTTPGet = &v1.HTTPGetAction{
+			Path: probePath,
+			Port: intstr.FromString(pt.ListensOn()),
+		}
+		if webSpec != nil && webSpec.TLSConfig != nil && version.GTE(semver.MustParse("2.24.0")) {
+			handler.HTTPGet.Scheme = v1.URISchemeHTTPS
+		}
+		return handler
+	}
+
+	// The /-/ready handler returns OK only after the TSDB initialization has
+	// completed. The WAL replay can take a significant time for large setups
+	// hence we enable the startup probe with a generous failure threshold (15
+	// minutes) to ensure that the readiness probe only comes into effect once
+	// Prometheus is effectively ready.
+	// We don't want to use the /-/healthy handler here because it returns OK as
+	// soon as the web server is started (irrespective of the WAL replay).
+	startupProbe := &v1.Probe{
+		ProbeHandler:     probeHandler("/-/ready"),
+		TimeoutSeconds:   probeTimeoutSeconds,
+		PeriodSeconds:    15,
+		FailureThreshold: 60,
+	}
+
+	livenessProbe := &v1.Probe{
+		ProbeHandler:     probeHandler("/-/healthy"),
+		TimeoutSeconds:   probeTimeoutSeconds,
+		PeriodSeconds:    5,
+		FailureThreshold: 6,
+	}
+
+	readinessProbe := &v1.Probe{
+		ProbeHandler:     probeHandler("/-/ready"),
+		TimeoutSeconds:   probeTimeoutSeconds,
+		PeriodSeconds:    5,
+		FailureThreshold: 3,
+	}
+
+	podAnnotations := map[string]string{}
+	podLabels := map[string]string{
+		"app.kubernetes.io/version": version.String(),
+	}
+	// In cases where an existing selector label is modified, or a new one is added, new sts cannot match existing pods.
+	// We should try to avoid removing such immutable fields whenever possible since doing
+	// so forces us to enter the 'recreate cycle' and can potentially lead to downtime.
+	// The requirement to make a change here should be carefully evaluated.
+	podSelectorLabels := map[string]string{
+		"app.kubernetes.io/name":       "prometheus",
+		"app.kubernetes.io/managed-by": "prometheus-operator",
+		"app.kubernetes.io/instance":   nc.BaseName(),
+		"prometheus":                   nc.BaseName(),
+		shardLabelName:                 fmt.Sprintf("%d", shard),
+		prometheusNameLabelName:        nc.BaseName(),
+	}
+	metadata := pt.GetPodMetadata()
+	if metadata != nil {
+		if metadata.Labels != nil {
+			for k, v := range metadata.Labels {
+				podLabels[k] = v
+			}
+		}
+		if metadata.Annotations != nil {
+			for k, v := range metadata.Annotations {
+				podAnnotations[k] = v
+			}
+		}
+	}
+
+	for k, v := range podSelectorLabels {
+		podLabels[k] = v
+	}
+
+	podAnnotations["kubectl.kubernetes.io/default-container"] = "prometheus"
```

Comment:
Maybe here the same from [comment](https://github.com/prometheus-operator/prometheus-operator/pull/4941/files#r936709304) also applies